components/incidents/createIncidents/CreateIncident.tsx

```diff
@@ -0,0 +1,363 @@
+            }, DURATION.long);
+            setIsSubmitClicked(false);
+    })
+    }
+
+    const toggleCheckBox = (e:React.BaseSyntheticEvent) =>{
+        const newComponentsAffected =  componentsAffected.map((item:ComponentObject) => {
+            if(item.id == e.target.name){
+                return {
+                    compName: item.compName,
+                    compType: item.compType,
+                    id: item.id,
+                    compId: item.compId,
+                    selected: !item.selected
+                }
+            }
+            else{
+                return item;
+            }
+        });
+        setComponentsAffected(
+           newComponentsAffected
+        );
+    }
+
+    const changeOption = (e:optionType, id:String) =>{
+        const newComponentsAffected =  componentsAffected.map((item:ComponentObject) => {
+            if(item.id.toString() == (id)){
+                return {
+                    compName: item.compName,
+                    compType: Number(e.option.id) + 1,
+                    id: item.id,
+                    compId: item.compId,
+                    selected: item.selected
+                }
+            }
+            else{
+                return item;
+            }
+        });
+        setComponentsAffected(
+           newComponentsAffected
+        );
+    }
+
+    const updateStatus =(e:React.BaseSyntheticEvent) =>{
+        setIncidentStatus(e.target.innerHTML);
+    }
+
+    const updateStatusBarOnClick = (e:SpecialEvent) => {
+        let percentage = 0;
+        let native = e.nativeEvent.offsetX;
+        if(!e.target.classList.contains("bar")){
+            console.log("YES!!!!");
+            let substractedTo = 0;
+            if(incidentStatus == "Identified"){
+                substractedTo = (66/100) * e.target.offsetWidth;
+            }
+            else if(incidentStatus == "Monitoring"){
+                substractedTo = (33/100) * e.target.offsetWidth;
+            }
+            percentage = ((e.nativeEvent.offsetX - substractedTo) * 100 / e.target.offsetWidth)
+        }
+        else{
+            percentage = (e.nativeEvent.offsetX * 100 / e.target.offsetWidth)
+            console.log("NO!!!!");
+        }
+        if(percentage < 16){
+            setIncidentStatus("Investigating");
+        }
+        else if(percentage < 50){
+            setIncidentStatus("Identified");
+        }
+        else if(percentage < 83){
+            setIncidentStatus( "Monitoring");
+        }
+        else{
+            setIncidentStatus("Resolved");
+        }
+    }
+
+    const formConstant = (
+    <>
+      <h2>Create Incident</h2>
+        <IncidentName value = {incidentName} handleNameChange = {(e:React.BaseSyntheticEvent) => handleNameChange(e)}/>
+        <InputStatus updateStatus = {(e:React.BaseSyntheticEvent) => updateStatus(e)} incidentStatus = {incidentStatus} updateStatusBarOnClick = {(event:SpecialEvent) => updateStatusBarOnClick(event)}/>
```

Comment:
updateStatusBarOnClick's logic should be inside InputStatus. Each form component should only take value and onChange method

Reply:
Done